test/stream.js

```diff
@@ -359,9 +359,32 @@ test('close socket on stream close', async function (t) {
     })
 })
 
-test('write non-buffer', async function (t) {
-  const u = new UDX()
-  const stream = u.createStream(1)
+test('write string', async function (t) {
+  t.plan(3)
+
+  const [a, b] = makeTwoStreams(t)
+
+  a
+    .on('data', function (data) {
+      t.alike(data, Buffer.from('hello world'))
+    })
+    .on('close', function () {
+      t.pass('a closed')
+    })
+    .end()
+
+  b
+    .on('close', function () {
+      t.pass('b closed')
+    })
+    .end('hello world')
+})
+
+test('write object', async function (t) {
+  const [a, b] = makeTwoStreams(t)
+
+  t.exception.all(() => a.write({ hello: 'world' }))
```

Comment:
imo we don't need to test this, the user should not do this.

Reply:
as in, if write throws, all bets are off (undefined behaivour)

Reply:
The issue, I guess, is that it doesn't throw; the process crashes with a native assertion error.

Reply:
that's fine also honestly - ie don't do that